src/unitycatalog/ai/databricks.py

```diff
@@ -620,3 +652,65 @@ def get_execute_function_sql_stmt(
     parts.append(")")
     statement = "".join(parts)
     return ParameterizedStatement(statement=statement, parameters=output_params)
+
+
+def get_execute_function_sql_command(function: "FunctionInfo", parameters: Dict[str, Any]) -> str:
+    from databricks.sdk.service.catalog import ColumnTypeName
+
+    sql_query = ""
+    if is_scalar(function):
+        sql_query += f"SELECT `{function.catalog_name}`.`{function.schema_name}`.`{function.name}`("
+    else:
+        sql_query += (
+            f"SELECT * FROM `{function.catalog_name}`.`{function.schema_name}`.`{function.name}`("
+        )
+
+    if parameters and function.input_params and function.input_params.parameters:
+        args: List[str] = []
+        use_named_args = False
+        for param_info in function.input_params.parameters:
+            if param_info.name not in parameters:
+                use_named_args = True
+            else:
+                arg_clause = ""
+                if use_named_args:
+                    arg_clause += f"{param_info.name} => "
+                param_value = parameters[param_info.name]
+                if param_info.type_name in (
+                    ColumnTypeName.ARRAY,
+                    ColumnTypeName.MAP,
+                    ColumnTypeName.STRUCT,
+                ):
+                    json_value_str = json.dumps(param_value)
+                    arg_clause += f"from_json('{json_value_str}', '{param_info.type_text}')"
+                elif param_info.type_name == ColumnTypeName.BINARY:
+                    if isinstance(param_value, bytes):
+                        param_value = base64.b64encode(param_value).decode("utf-8")
+                    # Use ubbase64 to restore binary values.
+                    arg_clause += f"unbase64('{param_value}')"
+                elif is_time_type(param_info.type_name.value):
+                    date_str = (
+                        param_value if isinstance(param_value, str) else param_value.isoformat()
+                    )
+                    arg_clause += f"'{date_str}'"
+                elif param_info.type_name == ColumnTypeName.INTERVAL:
+                    interval_str = (
+                        convert_timedelta_to_interval_str(param_value)
+                        if not isinstance(param_value, str)
+                        else param_value
+                    )
+                    arg_clause += interval_str
+                else:
+                    if param_info.type_name == ColumnTypeName.DECIMAL and isinstance(
+                        param_value, Decimal
+                    ):
+                        _logger.warning(
+                            f"Param {param_info.name} has decimal value {param_value}, it is converted to float "
+                            "for execution, please note that this conversion may lose precision."
+                        )
```

Comment:
Curious if we need this or not. This would theoretically be hit on every request for a given agent if it invokes a tool that is querying a table that contains this type... but builders of the app would be aware of this limitation. With the Decimal -> float casting, could we just leave this as a warning in the docs?

Reply:
Make sense!